From 0ff95108c3c07ba043d1a6bbfd696c20e0bf9002 Mon Sep 17 00:00:00 2001
From: Sergey Madaminov <sergey.madaminov@gmail.com>
Date: Tue, 14 Sep 2021 05:27:15 -0500
Subject: [PATCH] issue-61: fix handle pointer casting

---
 include/windows/unistd.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/windows/unistd.h b/include/windows/unistd.h
index 21cc56ff1..62ff90a3f 100644
--- a/include/windows/unistd.h
+++ b/include/windows/unistd.h
@@ -94,7 +94,7 @@ __inline long sysconf(int type)
 static __inline int
 rpl_isatty(int fd)
 {
-    HANDLE h = (HANDLE) _get_osfhandle(fd);
+    HANDLE h = (HANDLE)(INT_PTR) _get_osfhandle(fd);
     DWORD st;
     return (_isatty(STDOUT_FILENO)
             && h != INVALID_HANDLE_VALUE
-- 
2.32.0.windows.2

